BUILDING.md

```diff
@@ -31,4 +39,78 @@ The following trivial build script changes were made to successfully run `./grad
 * [job-scheduler#11](https://github.com/opensearch-project/job-scheduler/pull/11)
 * [anomaly-detection#18](https://github.com/opensearch-project/anomaly-detection/pull/18)
 
-We plan to remove the need for this work-around via [OpenSearch#581](https://github.com/opensearch-project/OpenSearch/issues/581).
\ No newline at end of file
+We plan to remove the need for this work-around via [OpenSearch#581](https://github.com/opensearch-project/OpenSearch/issues/581).
+
+## Developing new Plugins for OpenSearch
+
+All OpenSearch Plugins must contain the `plugin-descriptor.properties` file. The fields contained in the `plugin-descriptor.properties` file are mentioned [here](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/resources/plugin-descriptor.properties). If using the gradle build system, these fields can be specified in the `build.gradle` file and the build system will create the properties file for us.
```

Comment:
Make a section called `plugin.descriptor.properties`, another called `build.gradle`, etc. Don't make a section out of Example:

Reply:
Updated.